src/main/java/pl/schoolmanager/controller/StudentController.java

```diff
@@ -82,13 +84,15 @@ public String newStudentFromUserPost(@Valid @ModelAttribute Student student, Bin
 		HttpSession s = SessionManager.session();
 		s.setAttribute("thisStudent", student);
 		s.setAttribute("thisSchool", student.getSchool());
-		return "redirect:/studentView/division";
+		return "redirect:/studentView/";
 	}
 	
 	// Managing exisitng student role
 	@GetMapping("/userStudent")
 	public String StudentFromUser(Model m) {
 		m.addAttribute("student", new Student());
+		HttpSession s = SessionManager.session();
+		s.invalidate();
```

Comment:
You invalidate session here. What if this session contains another valuable information?